src/components/forms/ClusterSpecificInput.tsx

```diff
@@ -0,0 +1,136 @@
+import { FC, Fragment, useEffect, useRef, useState } from "react";
+import { CheckboxInput, Input } from "@canonical/react-components";
+import ResourceLink from "components/ResourceLink";
+import FormEditButton from "components/FormEditButton";
+import { ClusterSpecificValues } from "components/ClusterSpecificSelect";
+
+interface Props {
+  id: string;
+  isReadOnly: boolean;
+  onChange: (value: ClusterSpecificValues) => void;
+  toggleReadOnly: () => void;
+  memberNames: string[];
+  values?: ClusterSpecificValues;
+  canToggleSpecific?: boolean;
+  isDefaultSpecific?: boolean;
+  clusterMemberLinkTarget?: (member: string) => string;
+  disabled?: boolean;
+  helpText?: string;
+}
+
+const ClusterSpecificInput: FC<Props> = ({
+  values,
+  id,
+  isReadOnly,
+  memberNames,
+  onChange,
+  toggleReadOnly,
+  canToggleSpecific = true,
+  isDefaultSpecific = null,
+  clusterMemberLinkTarget = () => "/ui/cluster",
+  disabled = false,
+  helpText,
+}) => {
+  const [isSpecific, setIsSpecific] = useState<boolean | null>(
+    isDefaultSpecific,
+  );
+  const firstValue = Object.values(values ?? {})[0];
+  const isUserActionRef = useRef(false);
+
+  useEffect(() => {
+    if (isSpecific === null && values) {
+      const newDefaultSpecific = !Object.values(values).every(
+        (item) => item === Object.values(values)[0],
+      );
```

Comment:
```suggestion
      const newDefaultSpecific = Object.values(values).some(
        (item) => item !== Object.values(values)[0],
      );
```
Maybe bit more straight forward of the same logic?